packages/devtools_app/lib/src/debugger/codeview.dart

```diff
@@ -298,15 +315,35 @@ class _CodeViewState extends State<CodeView>
                         Expanded(
                           child: LayoutBuilder(
                             builder: (context, constraints) {
-                              return Lines(
-                                constraints: constraints,
-                                scrollController: textController,
-                                lines: lines,
-                                pausedFrame: pausedFrame,
-                                searchMatchesNotifier:
-                                    widget.controller.searchMatches,
-                                activeSearchMatchNotifier:
-                                    widget.controller.activeSearchMatch,
+                              final num fileWidth = lines.fold(
+                                  0,
+                                  (widestWidth, line) => math.max(widestWidth,
+                                      CodeView.calculateLineWidth(line)));
+                              final boxWidth = math.max(
+                                  constraints.minWidth, constraints.maxWidth);
+
+                              return Scrollbar(
```

Comment:
"With this change: If the viewport starts out big enough that a horizontal scrollbar is not required, and then is resized smaller so that a scrollbar is now needed, no scrollbar will show up."

Did you mean this scrollbar's behavior is abnormal?

Reply:
Yes, it's abnormal. Let me try to capture some short videos, that might be easiest. Thanks!

Reply:
Thanks,could you also test with this change #87818?

Reply:
Hopefully this is helpful. Here is the behavior with the stable Flutter SDK (as you can see, the issue described in https://github.com/flutter/flutter/issues/75613 is present - a non-functioning horizontal scrollbar is shown when it is not needed):

![stable_scrollbar](https://user-images.githubusercontent.com/21270878/129266280-33911917-e082-4dc0-8052-f3a260968739.gif)

And now here is the behavior using Flutter from head (including https://github.com/flutter/flutter/pull/87818). I'm shrinking the viewport so that the horizontal scrollbar is necessary. The horizontal scrollbar briefly appears for a second, but as I keep shrinking it disappears. 

![scrollbar_fork](https://user-images.githubusercontent.com/21270878/129267064-8327c698-6f12-421c-85be-daa3d76d3477.gif)

Reply:
Did you see the overflowing assertion of the second gif on the top right corner?

Reply:
Hmm I think it's unrelated. Here is another file that doesn't need to be shrunk as much to need the scrollbar. As you can see, the horizontal scrollbar doesn't appear while shrinking (and there is no overflow assertion). I then switch tabs to reload the page, and on reload the scrollbar shows up:

![scrollbar_fork2](https://user-images.githubusercontent.com/21270878/129269701-1cac15a4-746c-4be4-9fdd-4c84df4bbb16.gif)

Reply:
Did we change the bar's configurations such as 'isalwaysshow' during the resizing?

Reply:
Nope, `isAlwaysShow` doesn't change (it's set to `true`). I haven't made any changes since my last commit (https://github.com/flutter/devtools/pull/3262/commits/1057cd965c753a334652b67369ca92e6bdcd43bf), so all the code should be visible.

Reply:
Thanks very much!
I will try to reproduce this problem locally. If it can be reproduced, I believe it will be resolved soon.

Reply:
Great, thank you so much! The [contributing docs](https://github.com/flutter/devtools/blob/b9950aa0c6af0e095f2c5b254f41a127c9481a70/CONTRIBUTING.md) should be update with instructions on running devtools locally.

Reply:
I think I found the root cause, and I will fix it on the master branch today.

Reply:
https://github.com/flutter/flutter/pull/88152 should fix this, @elliette, Could you help verfy this patch actually fix the issue? Thanks very much!

Reply:
Yay, it's working with the patch! Thanks so much for your help getting this resolved, I really appreciate it. I will leave a comment in https://github.com/flutter/flutter/pull/88152 as well confirming that this fixes the issue I was seeing.